main.go

```diff
@@ -46,58 +27,44 @@ func main() {
 	}
-	gitDirpath := path.Join(currentWorkingDirpath, gitDirname)
-	if _, err := os.Stat(gitDirpath); err != nil {
-		if os.IsNotExist(err); err != nil {
-			return stacktrace.NewError("No Git directory found, meaning this isn't being run from the root of a Git repo")
+	gitrepositoryDirectory := path.Join(currentWorkingDirectory, gitDirname)
+	if _, err := os.Stat(gitrepositoryDirectory); err != nil {
+		if os.IsNotExist(err) {
+			return stacktrace.Propagate(err, "An error occurred getting the git repositorysitory in this directory. This means that this binary is not being run from root of a git repositorysitory.")
 		}
-		return stacktrace.Propagate(err, "An unrecognized error occurred getting the Git directory at '%v'", gitDirpath)
 	}
-	repository, err := git.PlainOpen(currentWorkingDirpath)
+	repository, err := git.PlainOpen(currentWorkingDirectory)
 	if err != nil {
-		return stacktrace.Propagate(err, "Couldn't get a Git repo from the current repository")
+		return stacktrace.Propagate(err, "An error occurred while attempting to open the existing git repositorysitory.")
 	}
-	originRemote, err := repository.Remote(originRemoteName)
+
+	// Check local master branch exists
+	localMasterBranch, err := repository.Branch(masterBranchName)
 	if err != nil {
-		return stacktrace.Propagate(err, "An error occurred getting remote '%v' for repository; is the code pushed?", originRemoteName)
+		return stacktrace.Propagate(err, "Missing required '%v' branch locally. Please run 'git checkout %v'", masterBranchName, masterBranchName)
 	}
+	fmt.Printf("Local master branch: %+v\n", localMasterBranch)
 
-	config.LoadConfig()
-
-	// Fetch if needed
-	lastFetchedFilepath := path.Join(gitDirpath, lastFetchedFilename)
-	shouldFetch := determineShouldFetch(lastFetchedFilepath)
-	if shouldFetch {
-		if err := originRemote.Fetch(&git.FetchOptions{}); err != nil {
-			return stacktrace.Propagate(err, "An error occurred fetching from the remote repository")
-		}
-		currentUnixTimeStr := fmt.Sprint(time.Now().Unix())
-		if err := ioutil.WriteFile(lastFetchedFilepath, []byte(currentUnixTimeStr), lastFetchedFileMode); err != nil {
-			return stacktrace.Propagate(err, "An error occurred writing last-fetched timestamp '%v' to file '%v'", currentUnixTimeStr, lastFetchedFilepath)
-		}
+	// Check no staged or unstaged changes exist on the branch before release
+	worktree, err := repository.Worktree()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while trying to retrieve the worktree of the repository.")
+	}
+	currWorktreeStatus, err := worktree.Status()
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred while trying to retrieve the status of the worktree of the repository.")
+	}
+	isClean := currWorktreeStatus.IsClean()
+	if !isClean {
+		fmt.Printf("The branch contains modified files. Please ensure the working tree is clean before attempting to release. Currently the status is '%s'\n", currWorktreeStatus)
```

Comment:
This should actually return an error, else you'll end up with a 0 (success) exit code which could cause calling scripts to think this succeeded when it didn't